tools/authors/add.php

```diff
@@ -354,20 +354,25 @@ function isValidDate(year, month, day) {
     echo "<input type='hidden' name='author_id' value='$author_id'>\n";
 }
 
-function _var($bd, $name)
+// Unfortunately the result of this function depends on the variable being accessed
+// so we can't type it. The way forward would be to add some class to encapsulate a
+// birth/death date with some meta-data.
+/** @return mixed */
+function _var(string $bd, string $name)
 {
     $var = $bd . $name;
     global $$var;
     return $$var;
 }
-function echo_date_fields($bd)
+
+function echo_date_fields(string $bd): void
 {
     ?>
 <table class='no-border'>
 <tr><td><?php echo _('Month'); ?>:</td><td>
-<?php echo write_months_list($bd, _var($bd, 'month')); ?></td></tr>
```

Comment:
This needed to be removed as `write_month_list` (and `write_days_list` below) is void and already `echo` their content.